WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -394,7 +411,16 @@ def submitJobs( self ):
       return result
     tqDict['Platform'] = result['Value']
     tqDict['Site'] = self.sites
-    tqDict['Tag'] = []
+
+    # Get a union of all tags
+    tags = []
+    for queue in self.queueDict:
+      tags += self.queueDict[queue]['ParametersDict'].get( 'Tag', [] )
+    tqDict['Tag'] = list( set( tags ) )
+
+    # Add overall max values for all queues
+    tqDict.update( self.globalParameters )
+
     self.log.verbose( 'Checking overall TQ availability with requirements' )
     self.log.verbose( tqDict )
 
```

Comment:
@atsareg We did the tests today and still have a problem here: pilots can't be submitted to multi-core sites without proper multi-core tags for queues. The reason is since tqDict doesn't pass multi-core tags as #3398 did, as last discussions, these tags should be added in the Matcher. But here rpcMatcher.getMatchingTaskQueues which is being called here doesn't translate multi-core parameters in tqDict to the necessary tags for the queues. Therefore, we suggest that getMatchingTaskQueues need to add the translation of multi-core parameters into tags.